core/src/main/scala_2.10/macrocompat/bundle_2.10.scala

```diff
@@ -167,7 +167,7 @@ class BundleMacro[C <: Context](val c: C) {
       case None => (Nil, List(tq"_root_.scala.AnyRef"), Nil)
     }
 
-    val mods = Modifiers(mods0.flags|ABSTRACT)
+    val mods = Modifiers(mods0.flags|ABSTRACT, mods0.privateWithin, mods0.annotations)
```

Comment:
In the real, internal implementation that's also a whole "positions" things that would probably be good to preserve, but I don't know how to do that with the actual API, so I left it:

https://github.com/scala/scala/blob/v2.10.6/src/reflect/scala/reflect/internal/Trees.scala#L915-L919